tripy/nvtripy/frontend/tensor.py

```diff
@@ -223,11 +238,35 @@ def eval(self) -> runtime.MemRefValue:
                 f"Note: Tensor was evaluated while compiling here: {str_from_stack_info(self.trace_tensor.eval_stack_info)}",
                 mode="once",
             )
+        return self
 
-        return data
+    def tolist(self) -> Union[List, numbers.Number]:
+        """
+        Returns the tensor as a nested list. If the tensor is a scalar, returns a python number.
 
-    def tolist(self):
-        data_memref = self.eval()
+        Returns:
+            The tensor represented as a nested list or a python number.
+
+        .. code-block:: python
+        :linenos:
+
+            # doc: print-locals tensor_list
+            tensor = tp.ones((2, 2))
+            tensor_list = tensor.tolist()
+
+            assert tensor_list == np.ones((2, 2), dtype=np.float32).tolist()
+
+        .. code-block:: python
+        :linenos:
```

Comment:
Something happened with the formatting here and above (maybe it's just a display issue on GitHub). 

Also, I think we'll need a caption:
```py
:caption: Scalar
```